contrib/containerized-pbench/pbench_demo

```diff
@@ -12,22 +12,24 @@
 # here because we're not using the default image).
 #-
 shopt -s expand_aliases
-alias pbench=$(git rev-parse --show-toplevel)/contrib/containerized-pbench/pbench
+alias pbench="$(git rev-parse --show-toplevel)"/contrib/containerized-pbench/pbench
 
 FIOTEST=${PWD}/fiotest
 export PB_AGENT_PODMAN_OPTIONS="--pull newer -v ${FIOTEST}:/fiotest:z"
 export PB_AGENT_IMAGE_NAME=quay.io/pbench/pbench-agent-all-fedora-36:main
 
-mkdir -p ${FIOTEST}
+mkdir -p "${FIOTEST}"
 
 #+
 # Run the demo!
 #-
+pbench pbench-generate-token --output=/var/lib/pbench-agent/.token
 pbench pbench-register-tool-set light
 pbench pbench-list-tools
 pbench pbench-user-benchmark --config example-workload -- \
     fio --directory=/fiotest --name fio_test_file --direct=1 --rw=randread \
-        --bs=16k --size=100M --numjobs=16 --time_based --runtime=20s \
+        --bs=16k --size=100M --numjobs=8 --time_based --runtime=5s \
         --group_reporting --norandommap
-pbench pbench-generate-token --output=/var/lib/pbench-agent/.token
-pbench pbench-results-move --token=$(< /var/tmp/pbench/pbench-agent/run/.token)
+# Note that the token file location below is evaluated -outside- the container,
+# unlike in the pbench-generate-token command above.
+pbench pbench-results-move --token="$(< /var/tmp/"${USER}"/pbench-agent/run/.token)"
```

Comment:
Odd. I suppose this was in the `main` version, too, and I didn't notice: but why not just `"$(< /var/tmp/${USER}/pbench-agent/run/.token)"` without the extra quotes? Wouldn't that have exactly the same effect? 🤔

Reply:
There is a sequence of things which happens:
- `${USER}` is expanded
- the surrounding construct is then word-split (so, if `${USER}` contains word-breaking characters, then the `$<` won't work as expected)
- the contents of the referenced file are inserted into the command line after the equals sign
- word-splitting happens again, and if there are any word-breaking characters in the token value, then only the first half is supplied to the `--token` switch and the rest becomes one or more command line arguments

So, technically, we need both sets of quotes.  That said, `${USER}` is _very_ unlikely to contain any word-breaking characters, and the contents of the bearer token are unlikely to contain any either, but using the quotes is easier and less obtrusive than adding directives to shut the linter up.

The `$()` (formerly "back-ticks") is a weird syntactical hole:  the stuff inside the parens is evaluated as a full command, and then the stuff around them is evaluated; so the quotes inside the parens are completely independent of any quoting outside the parens, and the result has to be quoted to avoid potential problems from word-splitting.

Reply:
> The `$()` (formerly "back-ticks") is a weird syntactical hole: the stuff inside the parens is evaluated as a full command, and then the stuff around them is evaluated; so the quotes inside the parens are completely independent of any quoting outside the parens

Huh. That makes sense, but it sure looks weird.

Reply:
> it sure looks weird.

Indeed.